source/common/common/interval_value.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <algorithm>
+
+namespace Envoy {
+
+// Template helper type that represents a closed interval with the given minimum and maximum values.
+template <typename T, T MinValue, T MaxValue> struct Interval {
+  static_assert(MinValue <= MaxValue, "min must be <= max");
+  static constexpr T min_value = MinValue;
+  static constexpr T max_value = MaxValue;
+};
+
+// Utility type that represents a value of type T in the given interval.
+template <typename T, typename Interval> class ClosedIntervalValue {
```

Comment:
Remove the T template argument, use Interval::T instead.  Alternatively, add compile asserts to verify that T and Interval::T are the same type.

Reply:
They can't be; float template parameters aren't allowed until C++20.